Dynatrace/dynatrace-operator
# Reconcile and mount self-signed TLS certificate for EEC and OtelC

## Description

[K8S-10820](https://dt-rnd.atlassian.net/browse/K8S-10820)

- New util library for TLS certificate generation
- EEC reconciler now also reconciles the self-signed certificate (if needed, only when no `tlsRefName` is specified)
- Always mount TLS certificate both on EEC and OtelC

## How can this be tested?

- Deploy DynaKube with
```
spec:
  extensions:
    prometheus:
      enabled: true
  templates:
    extensionExecutionController:
      imageRef:
        repository: extk8sregistry.azurecr.io/eec/dynatrace-eec
        tag: <latest-tag>
  activeGate:
    tlsSecretName: dynakube-ag-tls ## you have to create this one manually
```
- Check that `dynakube-extensions-controller-tls` secret gets reconciled correctly (when spec.templates.extensionsExecutionController.tlsRefName` is NOT set)
- Check that tls certificate gets added as volume mount in EEC and OtelC StatefulSets
